bin/adhoc-scripts/fetchFromReqmgr.py

```diff
@@ -0,0 +1,260 @@
+                    except KeyError:
+                        val = None
+                    wflow['tmp'][key] = val
+            elif wflow['RequestType'] == 'StepChain':
+                for key in keySetToFetch:
+                    try:
+                        val = wflow['Step1'][key]
+                    except KeyError:
+                        val = None
+                    wflow['tmp'][key] = val
+            elif wflow['RequestType'] == 'Resubmission':
+                pass
+            else:
+                logger.warning("Found unexpected workflow of type: %s", wflow['RequestType'])
+
+            # Append the High level keys identifying the workflow as key/value pairs in wflow['tmp']:
+            for key in keySetHighLevel:
+                wflow['tmp'][key] = wflow[key]
+
+            # Recalculate the missing values:
+            if wflow['tmp']['SplittingAlgo'] == 'EventBased':
+                logger.info("Found a workflow with EventBased Splitting algorithm:")
+                totalEvents = int(wflow['tmp']['RequestNumEvents'] / wflow['tmp']['FilterEfficiency'])
+                lumisPerJob = math.ceil(wflow['tmp']['EventsPerJob']/wflow['tmp']['EventsPerLumi'])
+                totalLumis = math.ceil(totalEvents / wflow['tmp']['EventsPerLumi'])
+                totalEstimatedJobs = math.ceil(totalLumis/lumisPerJob)
+
+                wflow['tmp']['TotalInputEvents'] = totalEvents
+                wflow['tmp']['TotalInputLumis'] = totalLumis
+                wflow['tmp']['TotalEstimatedJobs'] = totalEstimatedJobs
+                wflow['tmp']['TotalInputFiles'] = 0
+                logger.info('\n%s', pformat(wflow['tmp']))
+
+            elif wflow['tmp']['SplittingAlgo'] == 'EventAwareLumiBased':
+                logger.info("Found a workflow with EventAwareLumiBased Splitting algorithm:")
+                keySetInputLists = {'LumiList', 'RunList', 'BlockList'}
+                for key in keySetInputLists:
+                    try:
+                        wflow['tmp'][key] = wflow[key]
+                    except KeyError:
+                        wflow['tmp'][key] = None
+
+                if any([wflow['tmp'][key] for key in keySetInputLists]):
+                    logger.ino("Found input Lists for th e current workflow. Continue")
+                    logger.info('\n%s', pformat(wflow['tmp']))
+                    continue
+
+                # Fetch all datasetFile info. The result should be a list of dictionaries (one per file) of the kind:
+                # [...
+                #  {'Lumis': {1: [209, 230, 21, 181, 25, 146, 232, 74, 191, 207]},
+                #   'NumberOfEvents': 9289,
+                #   'ValidFile': 1},
+                #  {'Lumis': {1: [220, 19, 130, 154, 229]},
+                #   'NumberOfEvents': 4645,
+                #   'ValidFile': 1},
+                # ...]
+                datasetFileInfo = [{key: file[key] for key in ['LogicalFileName', 'Lumis', 'NumberOfEvents', 'ValidFile']}
```

Comment:
why don't you simply replace all this logic by:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/Services/DBS/DBS3Reader.py#L405

? Thus, using the DBS `filesummaries` as initially suggested.

Reply:
This is indeed much better.  Thanks @amaltaro . I am switching to this method now.